File: Problem sets/pset6 (Some problem sets but in python)/mario/mario.py
from cs50 import get_int

while True:
    h = get_int("height: ")
    if h >= 1 and h <= 8:
        break
# prompt user for input for height value of a positive integer between 1 and 8, break out of loop once conditions are met

for row in range(h):
    for col in range(h):
        if (row + col < h - 1):
            print(" ", end="")
        else:
            print("#", end="")
# print blank space for any position whose row and column sum is less than input height - 1, example row position 0 plus column position 1 is less than input height 4 - 1, so print space
# print # for any position whose row and column sum is not less than input height - 1, example row position 2 plus column position 2 is not less than input height 4 - 1, so print #

    print()

File: Problem sets/Mario/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int h;
    do
    {
        h = get_int("Height: ");
    }

    // expression to prompt user to imput figure between 1 and 8

    while
    (h < 1 || h > 8);

    // Nested loop

    for (int row = 0; row < h; row++)
    {
        for (int col = 0; col < h; col++)
        {
            if (row + col < h - 1)
            printf(" ");
            else
            printf("#");
    // condition: if the sum of the row and column is less than the specified height minus 1 print space other wise print #. Using a grid approach
        }
        printf("\n");
    }
    //Print the corresponding pyramid
}
